resources/js/cart.js

```diff
@@ -0,0 +1,66 @@
+function postCartData(type, data) {
+  return fetch('/cart/' + type, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(data)
+  }).then(res => res.json());
+}
+
+const cartController = (function() {
```

Comment:
cartController is not a controller but just updating the view. Instead combine this function and below function to create a proper Class.